sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/client.py

```diff
@@ -840,6 +841,9 @@ def _client_ready(self):
             return False
         if self._link.get_state().value != 3:  # ATTACHED
             return False
+        # once the receiver client is ready/connection established, we set prefetch
+        if not self._link_credit:   # if prefetch = 0, set link_credit to 1 to keep receiving
+            self._link.link_credit = self._link_credit + 1
```

Comment:
Rather than adding to `self._link_credit like this - would it be easier to just keep `self._link_credit` at our desired value, but simply always pass in `0` at line 831?

Reply:
While your observation is very true - I'm wondering if that's actually a _good_ thing....
It would only be on initial start up that we stop applying any kind of prefetch - which would then come into effect only once receive is called for the first time.... after which we start using link credit to determine how many messages to fetch per iteration......
I dunnno - do we have any customer engagements that might indicate one way or the other with regards to prefetching messages on initial start up?
It seems that uamqp never did this regardless of what the prefetch value was?

Reply:
setting line 831 to 0 as discussed, and passing in prefetch + 1 in the pyamqp/uamqp transports